genrl/deep/agents/ppo1/ppo1.py

```diff
@@ -86,20 +92,29 @@ def __init__(
 
     def create_model(self):
         # Instantiate networks and optimizers
-        input_dim, action_dim, discrete, action_lim = get_env_properties(
-            self.env, self.network_type
-        )
-
-        self.ac = get_model("ac", self.network_type)(
-            input_dim,
-            action_dim,
-            self.layers,
-            "V",
-            discrete,
-            action_lim=action_lim,
-            activation=self.activation,
-        ).to(self.device)
+        if isinstance(self.network, str):
+            input_dim, action_dim, discrete, action_lim = get_env_properties(
+                self.env, self.network
+            )
 
+            self.ac = get_model("ac", self.network)(
+                input_dim,
+                action_dim,
+                self.layers,
+                "V",
+                discrete,
+                action_lim=action_lim,
+                activation=self.activation,
+            ).to(self.device)
+        else:
+            self.ac = self.network.to(self.device)
+
+        assert "actor" and "critic" in dir(
+            self.ac
+        ), "network must contain actor and critic attributes"
+        assert "get_value" and "get_action" in dir(
+            self.ac
+        ), "network must contain get_value and get_action methods"
```

Comment:
Move to else statement. Can remove `actor` and `critic` as they're sort of obvious. `BaseActorCritic` also has all 4 of these by default so maybe all the assert statements can be removed.

Reply:
I guess that can be done. It woudl then run into a type error. So yeah, I'll remove these assert statements from all the agents